cpp/include/Ice/MarshaledResult.h

```diff
@@ -9,38 +9,55 @@
 #include "Current.h"
 #include "OutputStream.h"
 
+namespace IceInternal
+{
+    class Incoming;
+}
+
 namespace Ice
 {
 
 /**
- * Base class for marshaled result structures, which are generated for operations having the
- * marshaled-result metadata tag.
+ * Base class for marshaled result structures, which are generated for operations having the marshaled-result metadata
+ * tag.
  * \headerfile Ice/Ice.h
  */
 class ICE_API MarshaledResult
 {
 public:
 
-    // TODO: this class needs a virtual destructor
+    MarshaledResult() = delete;
```

Comment:
The public API of the base class is extremely limited. Basically:
 - public destructor
 - public move constructor
 - public move assignment